### DISCUSSION TITLE
Using secp256r1 keys, specifically the cipher TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256

### DISCUSSION THREAD

**Discussion by thoro** [mitmproxy/mitmproxy#5539] [Intercept Everything!]
How would I make mitmproxy to offer this cipher for the client ssl handshake?

**Comment by mhils** [comment]
We use OpenSSL via the pyca/cryptography package under the hood. You first need to make sure the the OpenSSL version includes thie cipher (I think this should be the case here), and then modify [`ciphers_server`](https://docs.mitmproxy.org/dev/concepts-options/#ciphers_server). The default ciphers are listed [here](https://github.com/mitmproxy/mitmproxy/blob/bc370276a19c1d1039e7a45ecdc23c362626c81a/mitmproxy/addons/tlsconfig.py#L17-L44), but that only applies to TLS 1.0-1.2 because the OpenSSL APIs are crazy. Noone has contributed a patch for TLS 1.3 cipher selection yet.

**Reply by thoro** [reply]
How about the certificates / keys? Does mitmproxy have an option to auto create ecdsa keys with specific curves?
Do I need to create all the certificates by hand?

**Reply by mhils** [reply]
> Does mitmproxy have an option to auto create ecdsa keys with specific curves?

I don't think we do.
